www/js/survey/enketo/answer.js

```diff
@@ -35,23 +36,37 @@ angular.module('emission.survey.enketo.answer', [
    * }}
    */
 
-  const ENKETO_SURVEY_CONFIG_PATH = 'json/enketoSurveyConfig.json';
   const LABEL_FUNCTIONS = {
-    TripConfirmSurvey: (xmlDoc) => {
-      const modeStr = _getAnswerByTagName(xmlDoc, 'travel_mode');
-      const purposeStr = _getAnswerByTagName(xmlDoc, 'destination_purpose');
-
-      if (modeStr.includes('trip_not_valid') || purposeStr.includes('trip_not_valid')) {
-        return 'Trip not valid';
-      }
-
-      const purposes = purposeStr.split(' ').length;
-      const modes = modeStr.split(' ').length;
-      return `${purposes} purpose${purposes > 1 ? 's': ''}, ${modes} mode${modes > 1 ? 's': ''}`;
-    },
-    UserProfileSurvey: (xmlDoc) => {
-      return 'Answered';
-    },
+    UseLabelTemplate: (xmlDoc, name) => {
+
+      return _lazyLoadConfig().then(configSurveys => {
+
+        const config = configSurveys[name]; // config for this survey
+        const lang = $translate.use();
+        const labelTemplate = config.labelTemplate?.[lang];
+
+        if (!labelTemplate) return "Answered"; // no template given in config
+        if (!config.labelVars) return labelTemplate; // if no vars given, nothing to interpolate,
+        // so we return the unaltered template
+
+        // gather vars that will be interpolated into the template according to the survey config
+        const labelVars = {}
+        for (lblVar in config.labelVars) {
+          const fieldName = config.labelVars[lblVar].key;
+          let fieldStr = _getAnswerByTagName(xmlDoc, fieldName);
+          if (fieldStr == '<null>') fieldStr = null;
+          if (config.labelVars[lblVar].type == 'length') {
+            const fieldMatches = fieldStr?.split(' ');
+            labelVars[lblVar] = fieldMatches?.length || 0;
+          } else {
+            labelVars[lblVar] = fieldStr || '';
+          }
```

Comment:
I would throw an exception here instead of returning the fieldStr. Code that is not tested might as well not be there.
And we are not planning to support anything other than length right now.

Reply:
I did this in this commit: https://github.com/e-mission/e-mission-phone/pull/917/commits/9cab667c6503e68edc2d2b49a8b2937b4716f43a

<img width="1459" alt="image" src="https://user-images.githubusercontent.com/61334340/216093191-c994dbf5-697c-4443-afcd-35008eb66e09.png">